pages/restmail.py

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import json
+import time
+import requests
+
+
+def get_login_link(username, message_count=1, timeout=60):
+    username = username.partition('@restmail.net')[0]
+    end_time = time.time() + timeout
+    while (True):
+        response = requests.get(
+            'http://restmail.net/mail/%s' % username,
+            verify=False)
+        restmail = json.loads(response.content)
+        requests.delete(
```

Comment:
This might trigger issues. There is a case where:

* We query restmail for email
* We store the response content to the variable
* By the time we trigger the delete, new emails might have arrived and we are dropping them (keep in mind that it's possible that an HTTP request might take some time for various reasons)